docs/site/HasMany-relation.md

```diff
@@ -529,9 +531,107 @@ export class CustomerRepository extends DefaultCrudRepository {
   for a certain relation. e.g
   `customerRepository.inclusionResolvers.delete('orders')`
 
-{% include note.html content="
-Inclusion with custom scope:
-Besides specifying the relation name to include, it's also possible to specify additional scope constraints.
-However, this feature is not supported yet. Check our GitHub issue for more information:
-[Include related models with a custom scope](https://github.com/strongloop/loopback-next/issues/3453).
-" %}
+### More use cases - query multiple relations:
+
+It is possible to query several relations or nested include relations with
+custom scope. Once you have the inclusion resolver of each relation set up, the
+following queries would allow you traverse data differently:
+
+In our example, we have relations:
+
+- `Customer` _belongsTo_ an `Address` - denoted as `address`.
+- `Customer` _hasMany_ `Order`s - denoted as `orders`.
+- `Order` _hasMany_ `Manufacturer` - denoted as `manufacturers`.
+
+To query **multiple relations**, for example, return all customers including
+their orders and address, in Node API:
+
+```ts
+customerRepo.find({include: [{relation: 'orders'}, {relation: 'address'}]});
+```
+
+Equivalently, with url, you can do:
+
+```
+GET http://localhost:3000/customers?filter[include][0][relation]=orders&filter[include][1][relation]=address
+```
+
+This gives
+
+```ts
+[
+  {
+    id: 1,
+    name: 'Thor',
+    addressId: 3
+    orders: [
+      {name: 'Mjolnir', customerId: 1},
+      {name: 'Rocket Raccoon', customerId: 1},
+    ],
+    address:{
+          id: 3
+          city: 'Thrudheim',
+          province: 'Asgard',
+          zipcode: '8200',
+    }
+  },
+  {
+    id: 2,
+    name: 'Captain',
+    orders: [{name: 'Shield', customerId: 2}], // doesn't have a related address
+  },
+]
+```
+
+To query **nested relations**, for example, return all customers including their
+orders and include orders' manufacturers whose name is `Toys R Us`, in Node API:
+
+```ts
+customerRepo.find({
+  include: [
+    {
+      relation: 'orders',
+      scope: {
+        where: {name: 'ToysRUs'},
+        include: [{relation: 'manufacturers'}],
+      },
+    },
+  ],
+});
+```
+
+Equivalently, with url, you can do:
+
+```
+// need to fix this
+ GET http://localhost:3000/customers?filter[include][0][relation]=orders&filter[include][0][scope][0]filter[where][name]=ToysRUs
```

Comment:
This url filters`[name]=ToysRUs` on `orders`. But I don't know what the syntax to query nested inclusion is. 
I tried
```
http://localhost:3000/customers?filter[include][0][relation]=orders&filter[include][0][scope]filter[include][0][relation]= manufacturers
```
but it doesn't work. I need some help here, thanks!

Reply:
My recommendation is to use [`qs.stringify`](https://www.npmjs.com/package/qs#stringifying) to convert the filter object to a query string.

Inside `packages/rest`:
```
$ node
> require('qs').stringify({filter: {include:[{relation: 'orders', scope: {include:[{relation: 'manufacturers'}]}}]}})
'filter%5Binclude%5D%5B0%5D%5Brelation%5D=orders&filter%5Binclude%5D%5B0%5D%5Bscope%5D%5Binclude%5D%5B0%5D%5Brelation%5D=manufacturers'
```

url-decoded:

```
$ node
> decodeURI(require('qs').stringify({filter: {include:[{relation: 'orders', scope: {include:[{relation: 'manufacturers'}]}}]}}))
'filter[include][0][relation]=orders&filter[include][0][scope][include][0][relation]=manufacturers'
```

Reply:
Personally, I would show users how to encode the filter via JSON:

```js
encodeURIComponent(JSON.stringify(filter))
```

There is a subtle difference in the output of `encodeURI` and `encodeURIComponent`, one of them is not correctly encoding special characters that must be encoded in query string values. I hope I remember it correctly that `encodeURIComponent` is the function to use.

Reply:
I still couldn't find a good way to encode it.
- `qs.stringify` and `encodeURIComponent` give different result.

- I found something weird, for filter
```
{filter: {include:[{relation: 'orders', scope: {include:[{relation: 'manufacturers'}]}}]}}
```
The outer relation is encoded to `{relation:"outer"}` as expected. But the inner one is not correct: `{[relation: inner]}`, which is causing invalid filter problems. I wonder why...

Even I would like to leave the encode part to users, I still think it's better to have at least one simple and working nested inclusion syntax as an example in docs.